src/main/java/tr/ogu/edu/school/schoolapp/service/PaymentService.java

```diff
@@ -15,19 +18,40 @@ public class PaymentService {
 
 	private final PaymentRepository paymentRepository;
 
-	public List<Payment> getAllPayments() {
-		return paymentRepository.findAll();
+	public List<PaymentDto> getAllPayments() {
+		List<Payment> payments = paymentRepository.findAll();
+		return payments.stream().map(PaymentMapper::toPaymentDto).collect(Collectors.toList());
 	}
 
-	public Optional<Payment> getPaymentById(Long id) {
-		return paymentRepository.findById(id);
+	public PaymentDto getPaymentById(Long id) {
+		Payment payment = paymentRepository.findById(id)
```

Comment:
güvenlik için bu sorguya oturum açan kullanıcının da dahil edilmesi uygun olacaktır.

Reply:
Evet hocam, oturum açma işlemlerini tamamlayıp bu pra geri döneceğim.